crates/astria-core/src/protocol/genesis/v1alpha1.rs

```diff
@@ -0,0 +1,778 @@
+            ibc_relayer_addresses: ibc_relayer_addresses.iter().map(Address::to_raw).collect(),
+            native_asset_base_denomination: native_asset_base_denomination.to_string(),
+            ibc_parameters: Some(ibc_parameters.to_raw()),
+            allowed_fee_assets: allowed_fee_assets.iter().map(ToString::to_string).collect(),
+            fees: Some(fees.to_raw()),
+        }
+    }
+}
+
+impl TryFrom<raw::GenesisAppState> for GenesisAppState {
+    type Error = <Self as Protobuf>::Error;
+
+    fn try_from(value: raw::GenesisAppState) -> Result<Self, Self::Error> {
+        Self::try_from_raw(value)
+    }
+}
+
+impl From<GenesisAppState> for raw::GenesisAppState {
+    fn from(value: GenesisAppState) -> Self {
+        value.into_raw()
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct GenesisAppStateError(GenesisAppStateErrorKind);
+
+impl GenesisAppStateError {
+    fn accounts(source: AccountError) -> Self {
+        Self(GenesisAppStateErrorKind::Accounts {
+            source,
+        })
+    }
+
+    fn address_prefixes(source: AddressPrefixesError) -> Self {
+        Self(GenesisAppStateErrorKind::AddressPrefixes {
+            source,
+        })
+    }
+
+    fn address_does_not_match_base(source: Box<AddressDoesNotMatchBase>) -> Self {
+        Self(GenesisAppStateErrorKind::AddressDoesNotMatchBase {
+            source,
+        })
+    }
+
+    fn allowed_fee_assets(source: ParseDenomError) -> Self {
+        Self(GenesisAppStateErrorKind::AllowedFeeAssets {
+            source,
+        })
+    }
+
+    fn authority_sudo_address(source: AddressError) -> Self {
+        Self(GenesisAppStateErrorKind::AuthoritySudoAddress {
+            source,
+        })
+    }
+
+    fn fees(source: FeesError) -> Self {
+        Self(GenesisAppStateErrorKind::Fees {
+            source,
+        })
+    }
+
+    fn field_not_set(name: &'static str) -> Self {
+        Self(GenesisAppStateErrorKind::FieldNotSet {
+            name,
+        })
+    }
+
+    fn ibc_relayer_addresses(source: AddressError) -> Self {
+        Self(GenesisAppStateErrorKind::IbcRelayerAddresses {
+            source,
+        })
+    }
+
+    fn ibc_sudo_address(source: AddressError) -> Self {
+        Self(GenesisAppStateErrorKind::IbcSudoAddress {
+            source,
+        })
+    }
+
+    fn native_asset_base_denomination(source: ParseTracePrefixedError) -> Self {
+        Self(GenesisAppStateErrorKind::NativeAssetBaseDenomination {
+            source,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("failed ensuring invariants of {}", GenesisAppState::full_name())]
+enum GenesisAppStateErrorKind {
+    #[error("`accounts` field was invalid")]
+    Accounts { source: AccountError },
+    #[error("`address_prefixes` field was invalid")]
+    AddressPrefixes { source: AddressPrefixesError },
+    #[error("one or more addresses did not match the provided base prefix")]
```

Comment:
Just seems odd using plural when we only provide a single address in the variant.

Not a big deal though - I'd stick with `one or more addresses` over `given addresses`, since we're not giving more than one address in the error :)

Reply:
I changed it to `"one of the provided addresses ..."`:  7569ca74